modules/gdscript/gdscript_parser.cpp

```diff
@@ -2805,6 +2805,14 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 					}
 					check_block = check_block->parent_block;
 				}
+				// Try to find variable name shadowing checking from the actual class
+				for (ClassNode *class_iterator = p_block->parent_class; class_iterator; class_iterator = class_iterator->owner) {
```

Comment:
It's not needed to iterate in the owner classes, they are never used in the same context. Just check the current class.

Also these checks should be in `_check_function_types()`, otherwise a situation where the class variable is declared after the function won't be caught, like this one:

```gdscript
func shadowing():
    var shadow

var shadow
```